### PR TITLE
refactor(example-getting-started): sync code and docs

### PR DESCRIPTION
**Pull request by virkt25**
connected to #729 

Updates the tutorial and the code so they are in sync with each other and the code reflects what is generated as someone follows along the tutorial using `@loopback/cli`. 

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/example-getting-started/docs/controller.md [hunk 1]
```diff
@@ -94,29 +93,36 @@ verbs:
...
     }
     return await this.todoRepo.create(todo);
   }
 
   @get('/todo/{id}')
-  async findTodoById(
-    @param.path.number('id') id: number, 
-    @param.query.boolean('items') items?: boolean): Promise<Todo> {
+  async findTodoById(@param.path.number('id') id: number): Promise<Todo> {
+    id = +id;
```
Comment (shimks) [inline packages/example-getting-started/docs/controller.md]: Why is this needed?

File: packages/example-getting-started/docs/putting-it-together.md [hunk 2]
```diff
@@ -26,26 +26,31 @@ other artifacts and inject them into our application for use.
...
 /* tslint:disable:no-unused-variable */
-// Do not remove!
-// Class and Repository imports required to infer types in consuming code!
 // Binding and Booter imports are required to infer types for BootMixin!
 import {BootMixin, Booter, Binding} from '@loopback/boot';
 import {
   Class,
   Repository,
   RepositoryMixin,
+  juggler,
```
Comment (shimks) [inline packages/example-getting-started/docs/putting-it-together.md]: We also need a notice for these two imports, similar to the notice given for `Binding` and `Booter`

File: packages/example-getting-started/src/application.ts [hunk 3]
```diff
@@ -5,39 +5,51 @@
...
-    // See https://github.com/strongloop/loopback-next/issues/441
+  setupDatasources() {
+    // This will allow you to test your application without needing to
+    // use the "real" datasource!
     const datasource =
       this.options && this.options.datasource
         ? new DataSourceConstructor(this.options.datasource)
         : db;
-    this.dataSource(datasource);
+    this.bind('datasource').to(datasource);
```
Comment (shimks) [inline packages/example-getting-started/src/application.ts]: Do we have a reason for not using the `dataSource` sugar?

File: packages/example-getting-started/docs/juggler.md [hunk 4]
```diff
@@ -49,6 +54,15 @@ export class TodoApplication extends BootMixin(
       },
     };
   }
+
+  async start() {
+    await super.start();
+
+    const server = await this.getServer(RestServer);
+    const port = await server.get('rest.port');
```
Comment (b-admike) [inline packages/example-getting-started/docs/juggler.md]: I would suggest to be explicit about the type e.g.`const port = server.get<number>('rest.port')`

File: packages/example-getting-started/src/application.ts [hunk 5]
```diff
@@ -5,39 +5,51 @@
...
+    this.setupDatasources();
   }
 
-  // Helper functions (just to keep things organized)
-  setupDataSources() {
-    // TODO(bajtos) Automate datasource and repo registration via @loopback/boot
-    // See https://github.com/strongloop/loopback-next/issues/441
+  setupDatasources() {
+    // This will allow you to test your application without needing to
+    // use the "real" datasource!
```
Comment (b-admike) [inline packages/example-getting-started/src/application.ts]: `the` -> `a`

File: packages/example-getting-started/docs/repository.md [hunk 6]
```diff
@@ -8,8 +8,7 @@ layer.
 
 ### Create your repository
 
-Create another folder in `src` called `repositories` and inside of that folder,
-create two files:
+In the `src/repositories` directory create two files:
```
Comment (b-admike) [inline packages/example-getting-started/docs/repository.md]: I suggest to add a comma after directory

File: packages/example-getting-started/docs/controller.md (hunk 1 above)
Comment (virkt25) [reply packages/example-getting-started/docs/controller.md]:
```
// REST adapter does not coerce parameter values coming from string sources
    // like path & query. As a workaround, we have to cast the value to a number
    // ourselves.
    // See https://github.com/strongloop/loopback-next/issues/750
    id = +id;
```

That's from the source code. I'll update the instructions to include a note on that as well.

File: packages/example-getting-started/docs/juggler.md (hunk 4 above)
Comment (virkt25) [reply packages/example-getting-started/docs/juggler.md]: This is the default as to what gets generated when a user runs `lb4 app` and I would rather not confuse users by adding stuff in here which isn't generated for them. 

This can be updated once the CLI templates are updated as such.

File: packages/example-getting-started/docs/controller.md (hunk 1 above)
Comment (shimks) [reply packages/example-getting-started/docs/controller.md]: Can you leave a comment on #750 so that it can be tracked and removed once we do have validation?

File: packages/example-getting-started/docs/juggler.md (hunk 4 above)
Comment (b-admike) [reply packages/example-getting-started/docs/juggler.md]: ah okay; that sounds good.

File: packages/example-getting-started/src/application.ts (hunk 3 above)
Comment (virkt25) [reply packages/example-getting-started/src/application.ts]: No reason, I was just going with the original instructions. Updated.

File: packages/example-getting-started/docs/controller.md [hunk 7]
```diff
@@ -94,29 +93,35 @@ verbs:
...
     }
     return await this.todoRepo.create(todo);
   }
 
   @get('/todo/{id}')
-  async findTodoById(
-    @param.path.number('id') id: number, 
-    @param.query.boolean('items') items?: boolean): Promise<Todo> {
+  async findTodoById(@param.path.number('id') id: number): Promise<Todo> {
     return await this.todoRepo.findById(id);
```
Comment (shimks) [inline packages/example-getting-started/docs/controller.md]: do we need `+id` here as well?

File: packages/example-getting-started/docs/juggler.md (hunk 4 above)
Comment (raymondfeng) [reply packages/example-getting-started/docs/juggler.md]: Let's fix here 1st to be `const port = server.get<number>('rest.port')`.

File: packages/example-getting-started/docs/controller.md (hunk 7 above)
Comment (shimks) [reply packages/example-getting-started/docs/controller.md]: weird

Comment (virkt25) [reply packages/example-getting-started/docs/controller.md]: Agreed.

**Comment by virkt25** [comment]
`db.datasource.ts` was updated to remove the unused import BUT I can't change the `path.resolve` to match the instructions since that would cause the mono repo tests to fail. This is a case where we can't avoid a difference due to the monorepo imo.
